builtin/game/statbars.lua

```diff
@@ -139,7 +139,8 @@ end
 
 function core.hud_replace_builtin(hud_name, definition)
 	if type(definition) ~= "table" or
-			definition.hud_elem_type ~= "statbar" then
+			(definition.hud_elem_type ~= "statbar" and
+			definition.type ~= "statbar") then
 		return false
 	end
```

Comment:
should do this to make compat work correctly:
```lua
definition.type = definition.type or definition.hud_elem_type
definition.hud_elem_type = nil
```

Reply:
Overwriting the definition type would prevent warnings from appearing if the deprecated field is used, so I don't think this is a good idea.

However, you are right this contains a small problem. (I just blindly used what SmallJoker suggested.)
If `hud_elem_type = "statbar"` and for example `type = "image"` the function will still be executed, and the element type will be `"image"`.

It's fixed now. I hope it's fine now.